src/Common/Enum/ReflectionEnum.php

```diff
@@ -54,12 +52,30 @@ public static function __callStatic(string $value, array $args = []): Enum
         return new static($constants[$value]);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public static function getAllowedValues(): array
+    {
+        $self = new ReflectionClass(static::class);
+
+        return $self->getConstants();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function equals(Enum $enum): bool
+    {
+        return $this->getValue() === $enum->getValue();
```

Comment:
Should we check class names too?
What happens if two different enums have the same value?

Reply:
mmm...
Maybe the better approach is to leave this as `abstract method`, so the implementation will figure out how to compare those objects.

Reply:
This can be the default behaviour. It is ever possible to override this method to accomplish with your need.

Reply:
@bigfoot90  yup, but the point is that being a "generic Value Object" we should enforce people to write their proper implementation.

I am not 100% sure about the equals method (with an Enum signature).
`public function equals(Enum $enum): bool` vs `public abstract function equals($enum): bool`.

Let me think about it 😄  Any feedback is appreciated.

Reply:
Let's go with: `public abstract function equals($enum): bool` in the implementation and `public  function equals($enum): bool` in the interface.

In this way, we'll be sure that the implementation can be type-hinted by userland with no restriction 😄 

Sorry for the delay 😄